eng/scripts/typespec_build.ps1

```diff
@@ -0,0 +1,113 @@
+#Requires -Version 7.0
+param(
+    [string]$filter,
+    [switch]$clean,
+    [switch]$vet,
+    [switch]$generate,
+    [switch]$skipBuild,
+    [switch]$cleanGenerated,
+    [switch]$format,
+    [switch]$tidy
+)
+
+. (Join-Path $PSScriptRoot .. common scripts common.ps1)
+
+function Process-TypeSpec ()
+{
+    $currentDirectory = Get-Location
+    
+    # Check for tsp-location.yaml first
+    if (-not (Test-Path "tsp-location.yaml")) {
+        Write-Host "##[error]This is not a TypeSpec-based SDK. No tsp-location.yaml file found in $currentDirectory"
+        return
+    }
+
+    if ($clean)
+    {
+        Write-Host "##[command]Executing go clean -v ./... in " $currentDirectory
+        go clean -v ./...
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
+    }
+
+    if ($cleanGenerated)
+    {
+        Write-Host "##[command]Cleaning auto-generated files in" $currentDirectory
+        (Get-ChildItem -recurse "*.go" | Where-Object { $_.Name -notlike '*_test.go' } | Select-String -Pattern "Code generated by Microsoft" | Select-Object -ExpandProperty path) | Remove-Item -Force
+    }
+
+    if ($generate)
+    {
+        Write-Host "##[command]Executing TypeSpec generation in " $currentDirectory
+        
+        # Install tsp-client if not already installed
+        Write-Host "##[command]Ensuring TypeSpec tooling is installed..."
+        npm list -g @azure-tools/typespec-client-generator-cli || npm install -g @azure-tools/typespec-client-generator-cli@v0.10.0
+        if ($LASTEXITCODE) {
+            Write-Host "##[error]Error installing TypeSpec tooling"
+            exit $LASTEXITCODE
+        }
```

Comment:
for env prerequisite, i don't believe we should ensure it here. for swagger case, we do not install `autorest` in the script, we just let cx to prepare the right env with doc.